python/tskit/trees.py

```diff
@@ -1582,10 +1652,45 @@ def nodes(self, root=None, order="preorder"):
             Unlike the :meth:`TreeSequence.nodes` method, this iterator produces
             integer node IDs, not :class:`Node` objects.
 
+        The currently implemented traversal orders are:
+
+            - 'preorder': starting at root, yield the current node, then recurse
+              and do a preorder on each child of the current node. See also `Wikipedia
+              <https://en.wikipedia.org/wiki/Tree_traversal#Pre-order_(NLR)>`__.
+            - 'inorder': starting at root, assuming binary trees, recurse and do
+              an inorder on the first child, then yield the current node, then
+              recurse and do an inorder on the second child. In the case of ``n``
+              child nodes (not necessarily 2), the first ``n // 2`` children are
+              visited in the first stage, and the remaining ``n - n // 2`` children
+              are visited in the second stage. See also `Wikipedia
+              <https://en.wikipedia.org/wiki/Tree_traversal#In-order_(LNR)>`__.
+            - 'postorder': starting at root, recurse and do a postorder on each
+              child of the current node, then yield the current node. See also
+              `Wikipedia
+              <https://en.wikipedia.org/wiki/Tree_traversal#Post-order_(LRN)>`__.
+            - 'levelorder' ('breadthfirst'): visit the nodes under root (including
+              the root) in increasing order of their depth from root. See also
+              `Wikipedia
+              <https://en.wikipedia.org/wiki/Tree_traversal\
+#Breadth-first_search_/_level_order>`__.
+            - 'timeasc': visits the nodes in order of increasing time, falling back to
+              increasing ID if times are equal.
+            - 'timedesc': visits the nodes in order of decreasing time, falling back to
+              decreasing ID if times are equal.
+            - 'minlex_postorder': a usual postorder has ambiguity in the order in
+              which children of a node are visited. We constrain this by outputting
+              a postorder such that the leaves visited, when their IDs are
+              listed out, have minimum `lexicographic order
+              <https://en.wikipedia.org/wiki/Lexicographical_order>`__ out of all valid
+              traversals. This traversal is useful for drawing multiple trees of
+              a ``TreeSequence``, as it leads to more consistency between adjacent
+              trees. Note that internal non-leaf nodes are not counted in
+              assessing the lexicographic order.
+
```

Comment:
I went with a bulleted list style. However, there is another style used in the documentation that I found: https://tskit.readthedocs.io/en/latest/python-api.html#tskit.TreeSequence.allele_frequency_spectrum. Is it fine to keep it as a bulleted list? Is the documentation too verbose?

Reply:
Better too much detail than too little. The wikipedia pages are very helpful for tree traversal order explanations, so I'm happy those are in there.

Reply:
This is excellent, thanks @brianzhang01!